### PR TITLE
✨DCS-1209: Add gender to incoming-moves resource

### PR DESCRIPTION
**Pull request by connormaglynn**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt [hunk 1]
```diff
@@ -35,7 +35,8 @@ class PrisonService(
         prisonNumber = it.offenderNo,
         date = it.movementDate,
         pncNumber = null,
-        isCurrentPrisoner = true
+        isCurrentPrisoner = true,
+        gender = null
```
Comment (connormaglynn) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt:39]: Left prison transfers as null, doesn't seem like we're getting any gender data from here, do we want to enrich this with BaSM data as well? Or is it completely separate?

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt:39]: yep, keep it separate.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/basm/Model.kt [hunk 2]
```diff
@@ -47,6 +47,16 @@ class Model {
...
+  )
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  data class Gender(
+    val key: String,
+    val title: String,
+    val description: String?,
+    val disabled_at: String?,
+    val nomis_code: String?
   )
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/basm/Model.kt:60]: did you have a go at mapping over to an enum at this point?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrival/Arrival.kt [hunk 3]
```diff
@@ -45,3 +48,8 @@ enum class LocationType {
   PRISON,
   OTHER
 }
+
+enum class Gender {
+  MALE,
+  FEMALE,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrival/Arrival.kt:54]: Might be better to add a comment here that we can only map across male and female, if BASM returns trans, then we pass back null as we need to explicitly capture gender information.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrival/Arrival.kt [hunk 4]
```diff
@@ -36,7 +36,10 @@ data class Arrival(
   val fromLocationType: LocationType,
 
   @Schema(description = "Is the arrival in custody according to NOMIS", example = "true")
-  val isCurrentPrisoner: Boolean = false
+  val isCurrentPrisoner: Boolean = false,
+
+  @Schema(description = "Gender of the arrival according to Book a Secure Move", example = "MALE")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrival/Arrival.kt:41]: Might be better to add a comment here that we can only map across male and female, if BASM returns trans, then we pass back null as we need to explicitly capture legal gender information.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**